src/libraries/Microsoft.Extensions.DependencyInjection/src/ServiceLookup/CallSiteFactory.cs

```diff
@@ -573,26 +572,23 @@ private static CallSiteResultCacheLocation GetCommonCacheLocation(CallSiteResult
             {
                 ServiceCallSite? callSite = null;
                 Type parameterType = parameters[index].ParameterType;
-                if (parameters[index].CustomAttributes != null)
+                foreach (var attribute in parameters[index].GetCustomAttributes(true))
```

Comment:
I assume getting all custom attributes is faster than 2 calls to get both `FromKeyedServicesAttribute` and `ServiceKeyAttribute` via the `GetCustomAttribute(Type attributeType` overload?

I suppose another option is to consoldate to one attribute (API change) if `GetCustomAttribute(Type attributeType`  is much faster than getting all attributes.

Reply:
I already tried to make `ServiceKeyAttribute` `sealed` and use `GetCustomAttribute(typeof(ServiceKeyAttribute)` (and only that, no trying to get the `FromKeyedServicesAttribute`), and I don't see any difference, at least when there is no attributes at all:

|    Method |        Job |                                                                 Toolchain |        Mode |     Mean |
|---------- |----------- |-------------------------------------------------------------------------- |------------ |---------:|
| Singleton | Job-GEPVBW | \runtime-fix\artifacts\bin\runtime\net8.0-windows-Release-x64\CoreRun.exe |     Dynamic | 1.658 us |
| Singleton | Job-OWDRFB |     \runtime\artifacts\bin\runtime\net8.0-windows-Release-x64\CoreRun.exe |     Dynamic | 1.753 us |
|           |            |                                                                           |             |          |
| Singleton | Job-GEPVBW | \runtime-fix\artifacts\bin\runtime\net8.0-windows-Release-x64\CoreRun.exe | Expressions | 1.616 us |
| Singleton | Job-OWDRFB |     \runtime\artifacts\bin\runtime\net8.0-windows-Release-x64\CoreRun.exe | Expressions | 1.634 us |
|           |            |                                                                           |             |          |
| Singleton | Job-GEPVBW | \runtime-fix\artifacts\bin\runtime\net8.0-windows-Release-x64\CoreRun.exe |      ILEmit | 1.650 us |
| Singleton | Job-OWDRFB |     \runtime\artifacts\bin\runtime\net8.0-windows-Release-x64\CoreRun.exe |      ILEmit | 1.646 us |
|           |            |                                                                           |             |          |
| Singleton | Job-GEPVBW | \runtime-fix\artifacts\bin\runtime\net8.0-windows-Release-x64\CoreRun.exe |     Runtime | 1.651 us |
| Singleton | Job-OWDRFB |     \runtime\artifacts\bin\runtime\net8.0-windows-Release-x64\CoreRun.exe |     Runtime | 1.651 us |